### PR TITLE
[jit] split TestLogging, TestDict, TestList

### PR DESCRIPTION
**Pull request by suo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#28038 [jit] split TestLogging, TestDict, TestList**
* #28037 [jit] split test_type_sharing

Differential Revision: [D17954441](https://our.internmc.facebook.com/intern/diff/D17954441)

### CODE REVIEW

File: test/jit/test_logging.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import os
+import sys
+
+import torch
+
+# Make the helper files in test/ importable
+pytorch_test_dir = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
```
Comment (eellison) [inline test/jit/test_logging.py]: Could you test that this is still runnable in Fbcode ? As far as I recall this doesn't work with buck, and it's a reason why onnx tests are not currently runnable in fbcode

Comment (suo) [reply test/jit/test_logging.py]: it is still runnable from fbcode. It's not hard to set up, just no one did it for onnx

**Review by driazati — COMMENTED** [review]
As long as we're splitting stuff why don't we move all of the test classes in `test_jit.py` into their own file? There's a semi-working script [here](https://github.com/pytorch/pytorch/pull/27624/files#diff-78aab5b4217aeaed920efab0c5f5f5c8R20225) to do it

**Comment by suo** [comment]
> As long as we're splitting stuff why don't we move all of the test classes in test_jit.py into their own file? There's a semi-working script here to do it

I'm moving a few over at a time manually, but that's the end state I think (and then we need to try splitting TestScript into sane chunks)

**Review by jamesr66a — APPROVED** [review]
stamp

**Pull request closed without merging**